lib/poxa/channels_handler.ex

```diff
@@ -74,16 +88,23 @@ defmodule Poxa.ChannelsHandler do
       end
   end
 
-  defp index(req, state) do
-    channels =
-      Channel.all
-        |> Enum.filter_map(
-          fn channel ->
-            Channel.presence? channel
-          end,
-          fn channel ->
-            {channel, [user_count: PresenceChannel.user_count(channel)]}
-          end)
+  defp index(filter, attributes, req, state) do
+    channels = channels(filter, attributes)
     {JSEX.encode!(channels: channels), req, state}
   end
+
+  defp channels(filter, attributes) do
```

Comment:
I usually prefer to use for comprehensions instead of using `Enum` whenever I can. I would write it like this:

```elixir
    for channel <- Channel.all, filter_channel(channel, filter) do
      {channel, mount_attribute_list(attributes, channel)}
    end
```